### DISCUSSION TITLE
How to search substring text of json value

### DISCUSSION THREAD

**Discussion by cpapad** [provectus/kafka-ui#3077] [Q&A] [answered]
Hello everyone!
We are evaluating `kafka-ui` to switch from another solution and so far we are very happy.
One thing though we can figure out or find it documented is how we can query for a substring of a json field value.
We have seen [this](https://github.com/provectus/kafka-ui/discussions/2428#discussioncomment-3396524) and managed to make multiple queries and filters but no luck till now managing to search and retrieve a message by a substring of a json filed value.

Example: A message containing `"product":"Smart TV"` was successfully returned when searching for "Smart TV" but was not retrieved while searching for "Smart" or "TV".

Thank you in advance

**Comment by iliax** [comment, marked as answer]
Hello @cpapad,
`Search` should work as you expected - as usual substring match filter. 
Potentially there is a situation when you used `Add Filter` and `Search` simultaneously - that can lead to error state. Please try to remove filter with `x` and try to reenter search string into `Search` area. 

---
In next release we planning to redesign whole Messages page to make it more intuitive to use.

**Comment by cpapad** [comment]
thank you very much for the response and the help!
